### PR TITLE
[SYSTEMML-1795] Specify a set of GPUs to use for a given machine

### PR DESCRIPTION
**Pull request by nakul02**
Can specify:
a) -1 for all GPUs
b) a specific number of GPU
c) a comma separated list of GPUs
d) a range of GPUs

Ping @niketanpansare

### CODE REVIEW

**Comment by akchinSTC** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://sparktc.ibmcloud.com/jenkins/job/SystemML-PullRequestBuilder/1787/

File: src/main/java/org/apache/sysml/runtime/instructions/gpu/context/GPUContextPool.java [hunk 1]
```diff
@@ -142,6 +154,58 @@ public synchronized static void initializeGPU() throws DMLRuntimeException {
 	}
 
 	/**
+	 * Parses a string into a list. The string can be of these forms:
+	 * <b/>
```
Comment (niketanpansare) [inline src/main/java/org/apache/sysml/runtime/instructions/gpu/context/GPUContextPool.java]: @nakul02 can you fix the bold tag ?

**Comment by akchinSTC** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://sparktc.ibmcloud.com/jenkins/job/SystemML-PullRequestBuilder/1792/

**Comment by nakul02** [comment]
@niketanpansare - the tests seem to have passed. 
Will wait for your review.

**Comment by niketanpansare** [comment]
LGTM. Thanks @nakul02  :+1:

**Comment by nakul02** [comment]
Thanks, I shall merge.

**Pull request closed without merging**